kibana/analytics-scripting/analytics_script.py

```diff
@@ -0,0 +1,234 @@
+            "max_date": max_date,
+            "sessionLength": bucket["sessionLength"]["value"],
+            "companyId": source["companyId"],
+            "flowId": source["flowId"],
+            "eventMessage": "Interaction Response Time",
+        }
+
+        # creating the single bulk action along with the document, which is
+        # stored under "_source"
+        bulk_action = {
+            "_index": index_to_update,
+            "_type": "_doc",
+            "_id": bucket["key"]["agg"],
+            "_source": document,
+            "_op_type": "index",
+        }
+
+        bulk_actions.append(bulk_action)
+    # sending the bulk request to Elasticsearch
+    helpers.bulk(elasticsearch, bulk_actions)
+
+def process_composite_aggregation_data(query_result):
+    """
+    Obtains necessary data for sending the Bulk API request that updates the
+    appropriate indices with the new data. Then prepares for obtaining the next
+    num_composite_buckets buckets from the composite aggregation result.
+    """
+    buckets = query_result["aggregations"]["my_buckets"]["buckets"]
+    source = query_result["aggregations"]["hits"]["hits"]["hits"][0]["_source"]
+    create_documents_and_send_bulk_request(buckets, source)
+
+    # Composite aggregation queries only return a specified number of buckets,
+    # num_composite_buckets in our case. The ordering of the buckets in the
+    # composite query is done by the "sources" property. Each time a composite
+    # query is run, it will return a key idenitifying the last returned bucket in
+    # the composite aggregation ordering. This key is placed into the "after_key"
+    # variable in this script. That variable is then used in running subsequent
+    # composite aggregation queries, in which we are obtaining additional
+    # composite aggregation buckets that we could not get from previous queries.
+    # When we add, or update, the "after":{"agg":after_key} structure to the
+    # composite aggregation portion of the query, we are telling elasticsearch
+    # to return the next num_composite_buckets, as defined by the composite
+    # aggregation ordering, from the set of all composite aggregation buckets.
+
+    query_composite = query_avg_response_time["aggs"]["my_buckets"]["composite"]
+    after_key = query_result["aggregations"]["my_buckets"]["after_key"]["agg"]
+
+    if "after" in query_composite:
+        # Updating the bucket idenifier in the composite["after"] section
+        # of the query. This tells elasticsearch to return the next
+        # num_composite_buckets in the ordering of the buckets in the
+        # composite aggregation results, which appear after the bucket
+        # with key after_key.
+        query_composite["agg"] = after_key
+    else:
+        # The first time we run the query, the composite["after"] section
+        # does not exist, so we must create it.
+        query_composite["after"] = {"agg": after_key}
```

Comment:
The comments are helpful but I don't think the if statement is necessary. Can't this just be simplified to
```suggestion
    query_composite.setdefault("after", {})
    query_composite["after"] = {"agg": after_key}
```